src/core/geometry/qgsgeometry.h

```diff
@@ -231,6 +231,20 @@ class CORE_EXPORT QgsGeometry
      */
     double distance( const QgsGeometry &geom ) const;
 
+#ifndef SIP_RUN
+    //! Returns STL-style iterator pointing to the first vertex of the geometry
+    //! \since QGIS 3.0
+    QgsAbstractGeometry::vertex_iterator vertices_begin() const;
+
+    //! Returns STL-style iterator pointing to the imaginary vertex after the last vertex of the geometry
+    //! \since QGIS 3.0
+    QgsAbstractGeometry::vertex_iterator vertices_end() const;
+#endif
+
+    //! Returns Java-style iterator for traversal of vertices of the geometry
+    //! \since QGIS 3.0
+    QgsVertexIterator vertices() const;
```

Comment:
Would be great to have a short code snippet here or on the class QgsVertexIterator
Maybe just a copy of the python test.